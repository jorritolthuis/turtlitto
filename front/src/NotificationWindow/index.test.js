import React from "react";
import { NotificationTypes, NotificationWindow } from ".";
import renderer from "react-test-renderer";

describe("NotificationWindow", () => {
  it("shows a SUCCESS notification", () => {
    const component = renderer.create(
      <NotificationWindow
        notificationType={NotificationTypes.SUCCESS}
        onDismiss={() => {
          console.log("Notification dismissed");
        }}
        message="This is a very annoying and extremely long error message which is displayed in full and hopefully uses wrap correctly with extremelylongwordswhichareuninterruptedandwithoutspaceswhichevenfillstheentirebox."
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("shows a WARNING notification", () => {
    const component = renderer.create(
      <NotificationWindow
        notificationType={NotificationTypes.WARNING}
        onDismiss={() => {
          console.log("Notification dismissed");
        }}
        message="This is a very annoying and extremely long error message which is displayed in full and hopefully uses wrap correctly with extremelylongwordswhichareuninterruptedandwithoutspaceswhichevenfillstheentirebox."
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("shows a ERROR notification", () => {
    const component = renderer.create(
      <NotificationWindow
        notificationType={NotificationTypes.ERROR}
        onDismiss={() => {
          console.log("Notification dismissed");
        }}
        message="This is a very annoying and extremely long error message which is displayed in full and hopefully uses wrap correctly with extremelylongwordswhichareuninterruptedandwithoutspaceswhichevenfillstheentirebox."
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("shows a non identified notification", () => {
    const component = renderer.create(
      <NotificationWindow
        notificationType={NotificationTypes.SOMETHING}
        onDismiss={() => {
          console.log("Notification dismissed");
        }}
        message="This is a very annoying and extremely long error message which is displayed in full and hopefully uses wrap correctly with extremelylongwordswhichareuninterruptedandwithoutspaceswhichevenfillstheentirebox."
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
