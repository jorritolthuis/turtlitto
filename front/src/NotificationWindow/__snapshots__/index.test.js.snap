// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationWindow shows a ERROR notification 1`] = `
<div
  className="sc-bdVaJa hIljtb"
>
  <header
    className="sc-bwzfXH cRtiHF"
  >
    ERROR
  </header>
  <button
    className="sc-bxivhb bfImHh"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 fa-4x "
      color="black"
      data-icon="times"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <p
    className="sc-htpNat etKPoU"
  >
    This is a very annoying and extremely long error message which is displayed in full and hopefully uses wrap correctly with extremelylongwordswhichareuninterruptedandwithoutspaceswhichevenfillstheentirebox.
  </p>
  <div />
</div>
`;

exports[`NotificationWindow shows a SUCCESS notification 1`] = `
<div
  className="sc-bdVaJa ipStDv"
>
  <header
    className="sc-bwzfXH cRtiHF"
  >
    SUCCESS
  </header>
  <button
    className="sc-bxivhb bfImHh"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 fa-4x "
      color="black"
      data-icon="times"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <p
    className="sc-htpNat etKPoU"
  >
    This is a very annoying and extremely long error message which is displayed in full and hopefully uses wrap correctly with extremelylongwordswhichareuninterruptedandwithoutspaceswhichevenfillstheentirebox.
  </p>
  <div />
</div>
`;

exports[`NotificationWindow shows a WARNING notification 1`] = `
<div
  className="sc-bdVaJa hsqBxz"
>
  <header
    className="sc-bwzfXH cRtiHF"
  >
    WARNING
  </header>
  <button
    className="sc-bxivhb bfImHh"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 fa-4x "
      color="black"
      data-icon="times"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <p
    className="sc-htpNat etKPoU"
  >
    This is a very annoying and extremely long error message which is displayed in full and hopefully uses wrap correctly with extremelylongwordswhichareuninterruptedandwithoutspaceswhichevenfillstheentirebox.
  </p>
  <div />
</div>
`;

exports[`NotificationWindow shows a non identified notification 1`] = `
<div
  className="sc-bdVaJa gwudjO"
>
  <header
    className="sc-bwzfXH cRtiHF"
  />
  <button
    className="sc-bxivhb bfImHh"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 fa-4x "
      color="black"
      data-icon="times"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <p
    className="sc-htpNat etKPoU"
  >
    This is a very annoying and extremely long error message which is displayed in full and hopefully uses wrap correctly with extremelylongwordswhichareuninterruptedandwithoutspaceswhichevenfillstheentirebox.
  </p>
  <div />
</div>
`;
